client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { get, merge } from 'lodash';
+
+import { decorrelatedJitter as defaultDelay } from './delays';
+import { default as defaultPolicy } from './policies';
+
+const isGetRequest = request => 'GET' === get( request, 'method', '' ).toUpperCase();
+
+export const retryOnFailure = ( getDelay = defaultDelay ) => inboundData => {
+	const { nextError, originalRequest, store: { dispatch } } = inboundData;
+
+	// if the request came back successfully
+	// then we have no need to intercept it
+	// flush our records of it
+	if ( ! nextError ) {
+		return inboundData;
+	}
+
+	// otherwise check if we should try again
+	if ( ! isGetRequest( originalRequest ) ) {
```

Comment:
not sure @blowery. I'd be happy to hone in on our errors, but say we are creating a new resource and then getting its value. if our `GET` gets to the server first it will return a `404` even though the resource will be there imminently. in this out-of-order edge case the auto-retry will come through and take care of that for us